### PR TITLE
SNAPTX-3458 Address Deprecations from 5.1

### PR DESCRIPTION
**Pull request by lorman**
Closes https://snapsheettech.atlassian.net/browse/SNAPTX-3458

Relaxes the requirements for ActiveRecord to allow for upstream rails updates.
Relaxes requirements for statesman gem to allow for updates.
Fixes an ActiveRecord::Dirty deprecation (this is breaking for older versions). 
Created a new Major version of the gem to make sure we pay attention to the breaking changes.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+version: 2.1
```
Comment (kazuuchikata) [inline .circleci/config.yml:1]: Is this project not currently being deployed via CircleCI? Is that why this file is a new file?

**Review by kazuuchikata — COMMENTED** [review]
I'm nervous about approving this, not because I see anything wrong, but I don't really have context on this repo and how it deploys, etc. I think I would want Santiago or another person with context to take a look first.

File: .circleci/config.yml (hunk 1 above)
Comment (lorman) [reply .circleci/config.yml:1]: @kazuuchikata this is a gem that being used to run most of the state machine logic inside of snapsheet-tx. This circleCI will only run the test, we didn't have it running previously. 
There are a ton of test of the state machine inside of snapsheet-tx. The new version of the gem here, will need to also be updated in the snapsheet-tx repo, and those test will have to pass. I will probably update the Statesman gem inside Snapsheet-tx as part of different PRs to make sure we can fully test everything. Right now it's throwing deprecations and the one code fix is necessary for the rest of the plan.

**Review by santi-h — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (kazuuchikata) [reply .circleci/config.yml:1]: Ah that makes me feel better about CircleCI. Thanks for providing the context, and you're right these changes in and of themselves shouldn't do anything. Thanks!

**Pull request merged**